### ISSUE TITLE
Require sections that summarize various concerns (a11y, privacy, security)

### ISSUE THREAD

**Issue by halindrome** [w3c/specberus#465]
Recently there have been a number of discussions about ensuring that critical requirements that come from various cross-discipline working groups at the W3C are addressed in each spec.  These include I18N, A11Y, Security, Privacy.  There are likely others.

I wonder if it would make sense for pub rules to require (or at least encourage) sections on these important topics within the Introduction?  For example, a spec might say:
1.  Introduction
   ........

1.1 Privacy Considerations

In this specification, the XYZ and PDQ interfaces allow web applications to request some user information.  The access to that information is controlled by user privacy settings in user agents that support these interfaces.  Web applications that request and are granted access to this information are expected to protect it as they would any other sensitive data.  

1.2  Security Considerations

...

1.3 Accessibility Considerations

This specification has no user interface components or otherwise interacts with users, except in that the user agent may request access to certain data via user privacy settings.  The manipulation of those privacy settings should be accessible, but are outside the scope of this specification.

1.4 Internationalization Considerations

...

That sort of thing.  Specberus could check that the sections are present, and the working groups tasked with doing lateral monitoring would have some guidance as to whether they even need to pay attention to a given spec without having to fully review it.

Thoughts?  In particular @plehegar @marcoscaceres

**Comment by marcoscaceres** [comment]
I don't have a strong pref if we do it via Specberus and/or in ReSpec/Bikeshed. I'm happy with whatever so long as the rules are clear.

**Comment by tripu** [comment]
I think some of those sections won't be applicable to some specs, so we can't emit an error if we don't find them. And adding yet more fixed, informative messages to the output (those &ldquo;advice&rdquo; labels in blue) will be noisy (there are already too many, IMHO).

I share the concern, but I'm not sure this should be part of formal pubrules-checking.

**Comment by tripu** [comment]
So as I said, I think we should try to shorten the output of Specberus, not extend it with even more fixed messages&nbsp;&mdash;&nbsp;those mask the important ones, and users tend to ignore them.

What do @deniak and @plehegar think about this?

**Comment by deniak** [comment]
:+1: @tripu. It seems hard to make these sections a requirement checked by pubrules

**Comment by plehegar** [comment]
Some sections are called "Security and Privacy considerations". We don't have sections for a11y and i18n considerations and a good chunk of them won't need to. I'd suggest issuing a warning if you can't the words "Privacy" or "Security" in any of the section headers.

**Comment by wseltzer** [comment]
I support making these Warning-level messages, warning "the Director will look for these considerations at transitions."

**Comment by tripu** [comment]
To illustrate my point: the output [for a spec that passes pubrules OK](https://www.w3.org/pubrules/?url=https%3A%2F%2Fwww.w3.org%2FTR%2Fwebsub%2F&profile=WD&validation=recursive&noRecTrack=false&informativeOnly=false&echidnaReady=false&patentPolicy=pp2004) is already quite verbose, and we suspect that most users ignore *warnings* and *advice* (everything that is not a blocking error in red). It's bad user interaction when most of the output can be ignored most of the time; the UI cries wolf too often.

If you're still OK to add more boilerplate, I'll add it.

**Comment by plehegar** [comment]
re too many warnings? Is that really so? I see that specberus gives warning on HTML and CSS validation, linkchecking, and permissive license. That doesn't seem too much and we can probably remove the warning on the use of permissive licence now. Having a warning is no security/privacy section detected would be useful imho.

**Comment by deniak** [comment]
@tripu is refering to the *advice* items. It's basically a bunch of "SHOULD" rules that cannot be checked within pubrules. I believe the suggestion made by @halindrome goes in that category of rules and I'm pretty sure, no one pays attention to these rules.
That being said, if you still want to add that a "rule" that will simply display a generic sentence, I suggest to use the same rule for the 4 sections listed by @halindrome (maybe even update the WCAG advice currently shown).

**Comment by plehegar** [comment]
Just made a proposal to indicate that documents should have a privacy and security sections.

**Comment by halindrome** [comment]
@plehegar I think that is fine, but I maintain that A11Y and I18N are equally if not more important.

**Comment by plehegar** [comment]
@halindrome except that those don't have dedicated sections in the specifications as far as I know. We can change that but we'd rather need the buyin from APA and I18n before changing pubrules.

**Comment by halindrome** [comment]
Of course.  I raised this issue because APA specifically requested it, and I18N has requested it in the past.  The various groups that do horizontal reviews all run into the same problems.  All the time.   But sure, I will ask APA to make a formal request if you like.
